File: 8/8.19.swift
// [GOOD]
// 責務を分けましょう
// Swiftならそのままstructで処理も記述すればいい

struct Reservation {
    var id: Int
    func cancelReservation() {
        // idを使った予約キャンセル処理
    }
}

struct ViewSettings {
    func darkMode() {
    }
}

struct MailMagazine {
    func beginSendMail() {
        // mailMagazineを使ったメール配信開始処理
    }
}

File: 8/8.19.dart
// [GOOD]
// 責務を分けましょう

class Reservation {
  final int id;
  
  Reservation(this.id);
  
  void cancelReservation() {
    // idを使った予約キャンセル処理
  }
}

class ViewSettings {
  void darkMode() {
    // ダークモード表示への変更処理
  }
}

class MailMagazine {
  void beginSendMail() {
    // メール配信開始処理
  }
}

void main() {
  final reservation = Reservation(1);
  final viewSettings = ViewSettings();
  final mailMagazine = MailMagazine();
  
}